schema/schemaspec/schemahcl/hcl.go

```diff
@@ -17,6 +17,21 @@ import (
 	"github.com/zclconf/go-cty/cty"
 )
 
+// Default is a default implementation of schemaspec.Marshaler and schemaspec.Unmarshaler
+// for Atlas HCL documents.
+var Default = &defaultCodec{}
```

Comment:
Why it's needed?

Reply:
In an upcoming PR, I will add on the driver package a func 

```go
func UnmarshalSpec(data []byte, unmarshaler schemaspec.Unmarshaler, v interface{}) error {
   // .. 
}
```

That can be invoked like:
```go
s := schema.Schema{}
err := mysql.UnmarshalSpec([]byte(f), schemahcl.Default, &s)
```

Reply:
So maybe choose a better name, like `DefaultCodec`?

Reply:
yeah i tried not to export the name codec because i don't like too much in this context. Codec usually is used in streaming contexts. I also didn't want to export something name "MarshalUnmarshaler" 
wdyt about exporting 

```go
var (
  Marshaler schemaspec.Marshaler = &defaultCodec{}
  Unmarshaler schemaspec.Unmarshaler = &defaultCodec{}
) 
```

Reply:
`Marshaler` and `Unmarshaler` are used as interfaces in multiple standard packages. I wouldn't use them as variable names. 

I understand your comment regarding the word coded, but I suggested it since you used `decode` and `encode` inside the package. Let's chat about it for a few seconds and find a solution.